### PR TITLE
Firebird Database Support

### PR DESCRIPTION
**Pull request by andrevanzuydam**
I have added support for the Firebird database engine:
Firebird is not a schema based database engine, I have attempted to indicate this with the messages but this may not be an acceptable change for the core flyway engine.  The actual implementation works fine with migrate and terminators using set term x; works correctly. I have also run tests with clean up which remove stored procedures,generators and tables correctly. I would want to update the documentation to this module as well as I know the Firebird database engine well.  I would be happy to implement the Interbase engine as another commit which would lean on this.

### CODE REVIEW

File: flyway-commandline/pom.xml [hunk 1]
```diff
@@ -80,13 +80,10 @@
             <groupId>org.xerial</groupId>
             <artifactId>sqlite-jdbc</artifactId>
         </dependency>
-
```
Comment (axelfontaine) [inline flyway-commandline/pom.xml]: Please don't mess with whitespace lines like these. They are placeholders for keeping line numbers in sync between the community, pro and enterprise editions.

File: flyway-core/src/main/java/org/flywaydb/core/internal/database/Schema.java [hunk 2]
```diff
@@ -148,7 +148,7 @@ public void clean() {
         try {
             doClean();
         } catch (SQLException e) {
-            throw new FlywaySqlException("Unable to clean schema " + this, e);
+            throw new FlywaySqlException("Unable to clean schema" + this, e);
```
Comment (axelfontaine) [inline flyway-core/src/main/java/org/flywaydb/core/internal/database/Schema.java]: Please don't introduce formatting bugs.

**Comment by axelfontaine** [comment]
Thanks for the pull request. Please see my initial comments.

**Comment by andrevanzuydam** [comment]
Thanks, would you like me to revert and resubmit?

On Jan 7, 2018 6:49 PM, "Axel Fontaine" <notifications@github.com> wrote:

Thanks for the pull request. Please see my initial comments.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/flyway/flyway/pull/1888#issuecomment-355835733>, or mute
the thread

**Comment by andrevanzuydam** [comment]
Going to fix this up

**Pull request closed without merging**